app/src/main/java/net/squanchy/schedule/ScheduleService.java

```diff
@@ -31,22 +32,26 @@
     private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd");
 
     private final FirebaseDbService dbService;
+    private final FirebaseAuthService authService;
     private final EventRepository eventRepository;
 
-    ScheduleService(FirebaseDbService dbService, EventRepository eventRepository) {
+    ScheduleService(FirebaseDbService dbService, FirebaseAuthService authService, EventRepository eventRepository) {
         this.dbService = dbService;
+        this.authService = authService;
         this.eventRepository = eventRepository;
     }
 
     public Observable<Schedule> schedule() {
-        final Observable<FirebaseDays> daysObservable = dbService.days();
-
-        return eventRepository.events()
-                .map(groupEventsByDay())
-                .withLatestFrom(daysObservable, combineSessionsById())
-                .map(sortPagesByDate())
-                .map(sortEventsByStartDate())
-                .subscribeOn(Schedulers.io());
+        return authService.signInAndObserve(userId -> {
+            final Observable<FirebaseDays> daysObservable = dbService.days();
```

Comment:
probably at some point it was made final because it was used inside an anonymous method, I'll see what happens if I remove it